examples/escrow/tests/escrow.js

```diff
@@ -0,0 +1,207 @@
+          escrowAccount: escrowAccount.publicKey,
+          tokenProgram: TOKEN_PROGRAM_ID,
+          rent: anchor.web3.SYSVAR_RENT_PUBKEY,
+        },
+        instructions: [
+          await program.account.escrowAccount.createInstruction(escrowAccount),
+        ],
+        signers: [escrowAccount],
+      }
+    );
+
+    // Get the PDA that is assigned authority to token account
+    const [_pda, _nonce] = await anchor.web3.PublicKey.findProgramAddress(
+      [Buffer.from(anchor.utils.bytes.utf8.encode("escrow"))],
+      program.programId
+    );
+
+    pda = _pda;
+
+    let _initializerTokenAccountA = await mintA.getAccountInfo(initializerTokenAccountA);
+
+    let _escrowAccount = await program.account.escrowAccount.fetch(
+      escrowAccount.publicKey
+    );
+
+    // Check that the new owner is the PDA
+    assert.ok(_initializerTokenAccountA.owner.equals(pda));
+
+    // Check that the values in the escrow account match what we expect
+    assert.ok(_escrowAccount.initializerKey.equals(provider.wallet.publicKey));
+    assert.ok(_escrowAccount.initializerAmount.toNumber() == initializerAmount);
+    assert.ok(_escrowAccount.takerAmount.toNumber() == takerAmount);
+    assert.ok(
+      _escrowAccount.initializerDepositTokenAccount.equals(initializerTokenAccountA)
+    );
+    assert.ok(
+      _escrowAccount.initializerReceiveTokenAccount.equals(initializerTokenAccountB)
+    );
+  });
+
+  it("Exchange escrow", async () => {
+    await program.rpc.exchange({
+      accounts: {
+        taker: provider.wallet.publicKey,
+        takerDepositTokenAccount: takerTokenAccountB,
+        takerReceiveTokenAccount: takerTokenAccountA,
+        pdaDepositTokenAccount: initializerTokenAccountA,
+        initializerReceiveTokenAccount: initializerTokenAccountB,
+        initializerMainAccount: provider.wallet.publicKey,
+        escrowAccount: escrowAccount.publicKey,
+        pdaAccount: pda,
+        tokenProgram: TOKEN_PROGRAM_ID,
+      },
+    });
+
+    let _takerTokenAccountA = await mintA.getAccountInfo(takerTokenAccountA);
+    let _takerTokenAccountB = await mintB.getAccountInfo(takerTokenAccountB);
+    let _initializerTokenAccountA = await mintA.getAccountInfo(initializerTokenAccountA);
+    let _initializerTokenAccountB = await mintB.getAccountInfo(initializerTokenAccountB);
+
+    // Check that the initializer gets back ownership of their token account
+    assert.ok(_takerTokenAccountA.owner.equals(provider.wallet.publicKey));
+
+    assert.ok(_takerTokenAccountA.amount.toNumber() == initializerAmount);
+    assert.ok(_initializerTokenAccountA.amount.toNumber() == 0);
+    assert.ok(_initializerTokenAccountB.amount.toNumber() == takerAmount);
+    assert.ok(_takerTokenAccountB.amount.toNumber() == 0);
+  });
+
+  let newEscrow = anchor.web3.Keypair.generate();
+
+  it("Initialize escrow and cancel escrow", async () => {
+    // Put back tokens into initializer token A account
+    await mintA.mintTo(
```

Comment:
@armaniferrante not sure why, but this call takes about 9.5 seconds for the promise to come back, any ideas?

Reply:
Definitely not. Sounds like a commitment setting.